packages/config/src/reader.ts

```diff
@@ -55,24 +63,25 @@ export class ConfigReader implements Config {
   }
 
   constructor(
-    private readonly data: JsonObject,
+    private readonly data: JsonObject | undefined,
     private readonly fallback?: ConfigReader,
+    private readonly prefix: string = '',
```

Comment:
Just a nit: while these are semantically correct, they are effectively three distinct unique ways of handling undefined as a parameter :) looks so inconsistent :)

Reply:
Yep but that's intentional. The 2nd param should not be omitted, the 3rd one can be omitted, while the 4th one can be omitted but has a default value